From 708eb1fe90a0418bb4241ed2ba49fc500e50802c Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:07:37 -0400
Subject: [powerpc] powernv: Add config option for hwpoisoning

Message-id: <1410545655-205645-229-git-send-email-dzickus@redhat.com>
Patchwork-id: 94587
O-Subject: [RHEL7 PATCH 228/626] powerpc/powernv: Add config option for hwpoisoning.
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit f2296a3d297d15f6df72ebb73a90af6b643c14dc
Author: Mahesh Salgaonkar <mahesh@linux.vnet.ibm.com>
Date:   Fri Nov 15 09:50:50 2013 +0530

    powerpc/powernv: Add config option for hwpoisoning.

    Add config option to enable generic memory hwpoisoning infrastructure for
    ppc64.

    Signed-off-by: Mahesh Salgaonkar <mahesh@linux.vnet.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/Kconfig b/arch/powerpc/Kconfig
index f95aadc..fcc8604 100644
--- a/arch/powerpc/Kconfig
+++ b/arch/powerpc/Kconfig
@@ -375,6 +375,12 @@ config ARCH_HAS_WALK_MEMORY
 config ARCH_ENABLE_MEMORY_HOTREMOVE
  def_bool y
 
+config PPC64_SUPPORTS_MEMORY_FAILURE
+ bool "Add support for memory hwpoison"
+ depends on PPC_BOOK3S_64
+ default "y" if PPC_POWERNV
+ select ARCH_SUPPORTS_MEMORY_FAILURE
+
 config KEXEC
  bool "kexec system call"
  depends on (PPC_BOOK3S || FSL_BOOKE || (44x && !SMP))
-- 
1.7.1